vision/camera/bag_file.py

```diff
@@ -95,29 +127,116 @@ def display_in_window(self, clipping=False):
 
             # render depth/color images
             depth_colormap = cv2.applyColorMap(
-                cv2.convertScaleAbs(depth_image, alpha=0.03),
-                cv2.COLORMAP_JET)
+                cv2.convertScaleAbs(depth_image, alpha=0.03), cv2.COLORMAP_JET
+            )
 
             if clipping:
                 bg_removed = np.where((depth_image_3d <= 0), grey_color, color_image)
                 images = np.hstack((bg_removed, depth_colormap))
             else:
                 images = np.hstack((color_image, depth_colormap))
 
-            cv2.namedWindow('Depth/Color Stream', cv2.WINDOW_NORMAL)
-            cv2.resizeWindow('Depth/Color Stream', self.width, int(self.height / 2))
-            cv2.imshow('Depth/Color Stream', images)
+            cv2.namedWindow("Depth/Color Stream", cv2.WINDOW_NORMAL)
+            cv2.resizeWindow("Depth/Color Stream", self.width, int(self.height / 2))
+            cv2.imshow("Depth/Color Stream", images)
 
             key = cv2.waitKey(1)
 
-            if key == ord('c'):
+            if key == ord("c"):
                 save_camera_frame(depth_image, color_image)
 
             # if pressed 'q' or escape (27) exit program
-            if key == ord('q') or key == 27 or cv2.getWindowProperty("Depth/Color Stream", 0) == -1:
+            if (
+                key == ord("q")
+                or key == 27
+                or cv2.getWindowProperty("Depth/Color Stream", 0) == -1
+            ):
                 cv2.destroyAllWindows()
                 break
 
-if __name__ == '__main__':
-    import sys
-    BagFile(720, 1080, 0, sys.argv[1]).display_in_window()
+    def save_as_img(self, folder_name: str = "new_set") -> None:
+        """
+        Iterates through the .bag file and saves color and depth frames as .jpg and .npy files respectively.
+        NOTE: Due to how .bag file is read, the number of frames saved may vary. Not every frame of the file is saved.
+        NOTE: Requires repeat to be false, or .bag file will be iterated over multiple times,
+              potentially resulting in repeat images.
+
+        Paramters
+        ---------
+        folder_name: str
+            The name of the folder to save the frames to.
+
+        Returns
+        -------
+        None.
+        """
+        DIRECTORY = os.path.join(folder_name)
```

Comment:
Wondering what might happen if `folder_name` is a string containing slashes. Don't think that edge case will be problematic though, as I feel like it will pretty much exclusively be basic strings.